### ISSUE TITLE
建议做表达式缓存

### ISSUE THREAD

**Issue by xxxxssss12** [killme2008/aviator#25]
功能的确很强大，只是效率上低了一丢丢，10000次计算a+b*c平均每次在200ms左右，生产环境下还是慢了些。。。simpleEL快太多了

**Comment by killme2008** [comment]
你怎么测试的。。怎么可能。单位是纳秒吧。

```java
    public static void main(String[] args) {
        long start = System.currentTimeMillis();
        Map<String, Object> env = new HashMap<String, Object>();
        env.put("a", 100);
        env.put("b", 101);
        env.put("c", 102);
        long sum = 0;
        for (int i = 0; i < 10000; i++)
            sum += (Long) AviatorEvaluator.execute("a+b*c", env);
        long end = System.currentTimeMillis();
        System.out.println((end - start) + "ms.");
        System.out.println(sum);
    }
```

1 万次调用 2.5 秒左右，平均 2.5 毫秒。

**Comment by killme2008** [comment]
启用  cache 模式

```java
sum += (Long) AviatorEvaluator.execute("a+b*c", env, true);
```

1万次只要 110 毫秒左右。

**Comment by lfkdsk** [comment]
Hi @killme2008 ~
1 million 110 ms efficiency does perform well, but I noticed that the default situation is using Eval mode, the general Eval model on the efficiency of this short expression should have a greater room for improvement.
I tried to implement an expression engine that supports compilation and interpretation of execution. Trying to use meta-cyclic explanations in the explanation to directly evaluate the performance of Ast's current performance.
![image](https://user-images.githubusercontent.com/11584398/31706408-7ca58f6c-b3ae-11e7-90e1-183021c478a2.png)
I think aviator can also try to optimize the AST / eval mode to make it better.
Thank you.
